packages/ui-components/src/hooks/useData.js

```diff
@@ -0,0 +1,37 @@
+/**
+ * Tupaia
+ * Copyright (c) 2017 - 2023 Beyond Essential Systems Pty Ltd
+ */
+
+import { useCallback, useState } from 'react';
+
+export const useData = fetcher => {
```

Comment:
This hooks is similar to the ones in react query, if built right this can be reused in many places